### ISSUE TITLE
Can't connect to argocd-repo-server after argo manages argo

### ISSUE THREAD

**Issue by eyearian** [argoproj/argo-cd#17069]
**Describe the bug**

Installing ArgoCD to an EKS cluster works without issue.  I can deploy the guestbook application without issue.  When I deploy an Argo Application to have ArgoCD manage ArgoCD, it loses all connectivity with the argocd-repo-server and seems to break connectivity to the argocd-server. Additionally, it breaks the guestbook application.  These are all running on the same node.  I have restarted all deployments after application deploy.

**To Reproduce**

`helm install my-argo-cd argo/argo-cd --version 5.53.0`

Once running, port forward, view the UI and deploy project.  
Project:
```
apiVersion: argoproj.io/v1alpha1
kind: AppProject
metadata:
  name: argo-poc
  namespace: argocd
  # Finalizer that ensures that project is not deleted until it is not referenced by any application
  # finalizers:
  #   - resources-finalizer.argocd.argoproj.io
spec:
  description: PoC Cluster
  destinations:
  - name: '*'
    namespace: '*'
    server: 'https://kubernetes.default.svc'
  clusterResourceWhitelist:
  - group: '*'
    kind: '*'
  sourceRepos:
  - '*'
 ```
Apply argocd application:
```
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: argocd
  namespace: argocd
  # finalizers:
  #   - resources-finalizer.argocd.argoproj.io
spec:
  project: argo-poc
  destination:
    namespace: argocd
    server: https://kubernetes.default.svc
  source:
    repoURL: https://argoproj.github.io/argo-helm
    chart: argo-cd
    targetRevision: 5.53.0
    helm:
      valuesObject:
        applicationSet:
          enabled: false
  syncPolicy:
    automated:
      prune: false
      selfHeal: true
```

**Expected behavior**

The argo app becomes the manager of the ArgoCD application.

**Screenshots**
![repo-server-error](https://github.com/argoproj/argo-cd/assets/7849887/5fe189bf-038b-4a06-9e3e-b1d02b894754)

This ip is the service of the argocd-repo-server

Once I restart the repo server I will lose connection to the UI and port forwarding stops working entirely.

**Version**

```shell
argocd: v2.9.3+6eba5be.dirty
  BuildDate: 2023-12-02T00:37:14Z
  GitCommit: 6eba5be864b7e031871ed7698f5233336dfe75c7
  GitTreeState: dirty
  GoVersion: go1.21.4
  Compiler: gc
  Platform: darwin/arm64
FATA[0000] Argo CD server address unspecified
```

**Logs**

application-controller:
```
level=info msg="Comparing app state (cluster: https://kubernetes.default.svc, namespace: argocd)" application=argocd/argocd
level=info msg="Skipping auto-sync: application status is Unknown" application=argocd/argocd
"Update successful" application=argocd/argocd
level=info msg="Reconciliation completed" application=argocd/argocd dedup_ms=0 dest-name= dest-namespace=argocd dest-server="https://kubernetes.default.svc" diff_ms=25 fields.level=2 git_ms=1 health_ms=0 live_ms=1 patch_ms=73 setop_ms=0 settings_ms=0 sync_ms=0 time_ms=123
```

repo-server:
```
level=info msg="finished unary call with code OK" grpc.code=OK grpc.method=Check grpc.service=grpc.health.v1.Health grpc.start_time="2024-02-01T15:46:21Z" grpc.time_ms=0.025 span.kind=server system=grpc
```

**Comment by todaywasawesome** [comment]
@eyearian have you tried it with manual sync and checked the diff before syncing?

**Comment by eyearian** [comment]
I figured it out 🤦 

It looks like installing the helm chart with the name `my-argo-cd` and renaming the cluster to `argocd` when applying the application breaks the cluster.  I noticed that in the manual sync diff when reinstalling argo. Everything else was the same except a checksum update. Once I installed the helm chart with the name `argo` and applied the application with the name `argocd` it worked after I restarted the redis pod.

Thanks so much for the quick response @todaywasawesome 
Hopefully this helps someone in the future.
